Messages/Messages/ChannelViewController.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  ChannelViewController.swift
+//  Messages
+//
+//  Created by Gustavo Mario Londoño Correa on 3/9/17.
+//  Copyright © 2017 JLainoG. All rights reserved.
+//
+
+import UIKit
+
+class ChannelViewController: UIViewController {
+  
+    @IBOutlet weak var channelsTable: UITableView!
+    @IBOutlet weak var newItemTxtField: UITextField!
+    
+    internal var channels: [ChannelProtocol] = []
+    
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        channelsTable.delegate = self
+        channelsTable.dataSource = self
+        channelsTable.register(UINib(nibName: "ChannelCell", bundle: nil), forCellReuseIdentifier: "cell")
+       
+        let service = ChannelService()
+        service.listChannels(){ channelsArray in
+            self.channels = channelsArray
+            self.channelsTable.reloadData()
+        }
+    }
+    
+    @IBAction func createChannel(_ sender: UIButton) {
+        let channel = PublicChannel(id:nil, name:newItemTxtField.text!)
+        ChannelService().createChannel(channel: channel)
+        channelsTable.reloadData()
+    }
+}
+
+extension ChannelViewController:UITableViewDataSource,UITableViewDelegate{
+    
+    
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+    
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        let headerTitles = ["Public Channels"]
+        if section < headerTitles.count {
+            return headerTitles[section]
```

Comment:
there is no need to create an array holding one element to then assign it when your table only has one section.